src/test/java/JP2ImageConverter/services/KakaduServiceTest.java

```diff
@@ -391,6 +415,74 @@ public void testKduCompressRotatedJpg() throws Exception {
         assertContains("2832x4256", output);
     }
 
+    @Test
+    public void testKduCompressYCbCrTif() throws Exception {
+        String mockedTif = String.valueOf(new File( tmpFolder + "/mockedImage.tif"));
+        Map<String, String> imageMetadata = Map.of(ColorFieldsService.PHOTOMETRIC_INTERPRETATION, "YCbCr");
+        ColorFieldsService colorFieldsService = mock(ColorFieldsService.class);
+        when(colorFieldsService.extractMetadataFields(anyString())).thenReturn(imageMetadata);
+        when(colorFieldsService.identifyType(anyString())).thenReturn("RGB");
+        ImagePreproccessingService imagePreproccessingService = mock(ImagePreproccessingService.class);
+        when(imagePreproccessingService.convertToTiff(anyString(), anyString())).thenReturn(mockedTif);
+        when(imagePreproccessingService.convertColorSpaces(anyString(), anyString())).thenReturn(mockedTif);
+
+        try (MockedStatic<CommandUtility> mockedStatic = Mockito.mockStatic(CommandUtility.class)) {
+            File mockedJp2 = new File(tmpFolder + "/mockedImage.jp2");
```

Comment:
```suggestion
            String mockedJp2String = tmpFolder.resolve("mockedImage.jp2").toString();
```
Then we can remove the conversions to strings later